### ISSUE TITLE
Pencil's binding to the cursor is inconvenient on even sizes

### ISSUE THREAD

**Issue by GordemProo** [aseprite/aseprite#2868]
If the pencil's size in an even number, it should be attached to the pixel grid with a half-pixel offset on each axis. That's how it is in present:
![brush-alignment-grid-wrong](https://user-images.githubusercontent.com/80538841/128377341-84ffff68-faa0-4526-8617-cec166b9305a.png)
And it should be like this:
![brush-alignment-grid-right](https://user-images.githubusercontent.com/80538841/128392122-be176d19-9a49-461d-8d4e-8f9985aca23b.png)
The black-and-white line is the previous, wrong position of the pencil.

In other words, the pencil square should always calculate a one-pixel-size square precisely in the center of him, and take a such a position for the cursor to be in this one-pixel square. (In present, the centers of these two squares do not fit, it is the top left corner that fits.)

Due to the thing described above it's hard to point the pencil “at once”.

**Comment by dacap** [comment]
Hi there 👋  I think this might be related to #723

**Comment by GordemProo** [comment]
@dacap
Yes, that is a very similar problem, but not exactly the same. Here it's probably the matter of how the pencil works, and there it is a matter of how Snap to Grid behaves together with the pencil, (and there's something wrong with Snap).
